src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -0,0 +1,338 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+from __future__ import annotations
+from concurrent.futures import (
+    Executor,
+)
+import os
+from pathlib import Path
+import sys
+import json
+from shlex import quote
+from logging import getLogger, LoggerAdapter
+import sysconfig
+from typing import Any, TYPE_CHECKING, Optional
+from dataclasses import asdict
+
+from deadline.job_attachments.asset_manifests import BaseAssetManifest
+from deadline.job_attachments.models import (
+    Attachments,
+    PathFormat,
+    JobAttachmentS3Settings,
+    ManifestProperties,
+    PathMappingRule,
+    JobAttachmentsFileSystem,
+)
+from deadline.job_attachments.os_file_permission import (
+    FileSystemPermissionSettings,
+    PosixFileSystemPermissionSettings,
+    WindowsFileSystemPermissionSettings,
+    WindowsPermissionEnum,
+)
+
+from openjd.sessions import (
+    LOG as OPENJD_LOG,
+    PathMappingRule as OpenjdPathMapping,
+    PosixSessionUser,
+    WindowsSessionUser,
+)
+from openjd.model.v2023_09 import (
+    EmbeddedFileTypes as EmbeddedFileTypes_2023_09,
+    EmbeddedFileText as EmbeddedFileText_2023_09,
+    Action as Action_2023_09,
+    StepScript as StepScript_2023_09,
+    StepActions as StepActions_2023_09,
+)
+from openjd.model import ParameterValue
+
+from ..session import Session
+from ...log_messages import SessionActionLogKind
+
+
+from .openjd_action import OpenjdAction
+
+if TYPE_CHECKING:
+    from concurrent.futures import Future
+    from ..session import Session
+    from ..job_entities import JobAttachmentDetails, StepDetails
+
+
+logger = getLogger(__name__)
+
+
+class SyncCanceled(Exception):
+    """Exception indicating the synchronization was canceled"""
+
+    pass
+
+
+class AttachmentDownloadAction(OpenjdAction):
+    """Action to synchronize input job attachments for a AWS Deadline Cloud job
```

Comment:
nit, for Deadline Cloud Session? Since a session has job and step-step dependency.